AwsEncryptionSDK/runtimes/rust/examples/keyring/ecdh/ephemeral_raw_ecdh_keyring_example.rs

```diff
@@ -0,0 +1,174 @@
+
+    // 5. Create the EphemeralPrivateKeyToStaticPublicKeyInput
+    let ephemeral_raw_ecdh_static_configuration_input =
+        EphemeralPrivateKeyToStaticPublicKeyInput::builder()
+            // Must be a DER-encoded X.509 public key
+            .recipient_public_key(public_key_recipient_utf8_bytes)
+            .build()?;
+
+    let ephemeral_raw_ecdh_static_configuration =
+        RawEcdhStaticConfigurations::EphemeralPrivateKeyToStaticPublicKey(ephemeral_raw_ecdh_static_configuration_input);
+
+    // 6. Create the Ephemeral Raw ECDH keyring.
+    let mpl_config = MaterialProvidersConfig::builder().build()?;
+    let mpl = mpl_client::Client::from_conf(mpl_config)?;
+
+    // Create the keyring.
+    // This keyring uses an ephemeral configuration. This configuration will always create a new
+    // key pair as the sender key pair for the key agreement operation. The ephemeral configuration can only
+    // encrypt data and CANNOT decrypt messages.
+    let ephemeral_raw_ecdh_keyring = mpl
+        .create_raw_ecdh_keyring()
+        .curve_spec(ecdh_curve_spec)
+        .key_agreement_scheme(ephemeral_raw_ecdh_static_configuration)
+        .send()
+        .await?;
+
+    // 7. Encrypt the data with the encryption_context
+
+    // A raw ecdh keyring with Ephemeral configuration cannot decrypt data since the key pair
+    // used as the sender is ephemeral. This means that at decrypt time it does not have
+    // the private key that corresponds to the public key that is stored on the message.
+    let plaintext = example_data.as_bytes();
+
+    let encryption_response = esdk_client.encrypt()
+        .plaintext(plaintext)
+        .keyring(ephemeral_raw_ecdh_keyring)
+        .encryption_context(encryption_context)
+        .send()
+        .await?;
+
+    let ciphertext = encryption_response
+                        .ciphertext
+                        .expect("Unable to unwrap ciphertext from encryption response");
+
+    // 8. Demonstrate that the ciphertext and plaintext are different.
+    // (This is an example for demonstration; you do not need to do this in your own code.)
+    assert_ne!(ciphertext, aws_smithy_types::Blob::new(plaintext),
+        "Ciphertext and plaintext data are the same. Invalid encryption");
+
+    println!("Ephemeral Raw ECDH Keyring Example Completed Successfully");
+
+    Ok(())
+}
+
+fn should_generate_new_ecc_key_pair_ephemeral_raw_ecdh() -> Result<bool, String> {
+    // If key already exists: do not overwrite existing key
+    if exists(EXAMPLE_ECC_PUBLIC_KEY_FILENAME_RECIPIENT) {
+        Ok(false)
+    }
+    // If file is not present, generate a new key pair
+    else {
+        Ok(true)
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+pub async fn test_encrypt_with_keyring() -> Result<(), crate::BoxError2> {
+    // Test function for encrypt using the Ephemeral Raw ECDH Keyring example
+    use crate::example_utils::utils;
+
+    encrypt_with_keyring(
```

Comment:
I would rather not change this because then I would have to change all the other functions in all examples, and this is the only function in this file so this should be fairly obvious.